sycl/unittests/assert/assert.cpp

```diff
@@ -0,0 +1,393 @@
+                                             pi_kernel_group_info param_name,
+                                             size_t param_value_size,
+                                             void *param_value,
+                                             size_t *param_value_size_ret) {
+  if (param_name == PI_KERNEL_GROUP_INFO_COMPILE_WORK_GROUP_SIZE) {
+    if (param_value_size_ret) {
+      *param_value_size_ret = 3 * sizeof(size_t);
+    } else if (param_value) {
+      auto size = static_cast<size_t *>(param_value);
+      size[0] = 1;
+      size[1] = 1;
+      size[2] = 1;
+    }
+  }
+
+  return PI_SUCCESS;
+}
+
+static pi_result redefinedEnqueueKernelLaunch(pi_queue, pi_kernel, pi_uint32,
+                                              const size_t *, const size_t *,
+                                              const size_t *LocalSize,
+                                              pi_uint32 N, const pi_event *Deps,
+                                              pi_event *RetEvent) {
+  int *Ret = new int[1];
+  *Ret = KernelLaunchCounter++;
+  // This output here is to reduce amount of time requried to debug/reproduce a
+  // failing test upon feature break
+  printf("Enqueued %i\n", *Ret);
+
+  if (PauseWaitOnIdx == *Ret) {
+    // It should be copier kernel. Check if it depends on user's one.
+    EXPECT_EQ(N, 1U);
+    int EventIdx = reinterpret_cast<int *>(Deps[0])[0];
+    EXPECT_EQ(EventIdx, 0);
+  }
+
+  *RetEvent = reinterpret_cast<pi_event>(Ret);
+  return PI_SUCCESS;
+}
+
+static pi_result redefinedEventsWait(pi_uint32 num_events,
+                                     const pi_event *event_list) {
+  // there should be two events: one is for memory map and the other is for
+  // copier kernel
+  assert(num_events == 2);
+
+  int EventIdx1 = reinterpret_cast<int *>(event_list[0])[0];
+  int EventIdx2 = reinterpret_cast<int *>(event_list[1])[0];
+  // This output here is to reduce amount of time requried to debug/reproduce a
+  // failing test upon feature break
+  printf("Waiting for events %i, %i\n", EventIdx1, EventIdx2);
+  return PI_SUCCESS;
+}
+
+static pi_result
+redefinedMemBufferCreate(pi_context context, pi_mem_flags flags, size_t size,
+                         void *host_ptr, pi_mem *ret_mem,
+                         const pi_mem_properties *properties = nullptr) {
+  *ret_mem = nullptr;
+  return PI_SUCCESS;
+}
+
+static pi_result redefinedMemRelease(pi_mem mem) { return PI_SUCCESS; }
+
+static pi_result redefinedKernelSetArg(pi_kernel kernel, pi_uint32 arg_index,
+                                       size_t arg_size, const void *arg_value) {
+  return PI_SUCCESS;
+}
+
+static pi_result redefinedEnqueueMemBufferMap(
+    pi_queue command_queue, pi_mem buffer, pi_bool blocking_map,
+    pi_map_flags map_flags, size_t offset, size_t size,
+    pi_uint32 num_events_in_wait_list, const pi_event *event_wait_list,
+    pi_event *RetEvent, void **RetMap) {
+  int *Ret = new int[1];
```

Comment:
It looks like a memory leak. Can we avoid this?

Reply:
I don't think it's really relevant as the process will abort anyway.